lib/cisco_node_utils/itd_service.rb

```diff
@@ -0,0 +1,518 @@
+    end
+
+    def access_list
+      config_get('itd_service', 'access_list', @get_args)
+    end
+
+    def access_list=(val)
+      if val
+        @set_args[:state] = ''
+        @set_args[:al] = val
+        config_set('itd_service', 'access_list', @set_args)
+      else
+        @set_args[:state] = 'no'
+        @set_args[:al] = access_list
+        config_set('itd_service', 'access_list', @set_args) if
+        access_list
+      end
+      set_args_keys_default
+    end
+
+    def default_access_list
+      config_get_default('itd_service', 'access_list')
+    end
+
+    def device_group
+      config_get('itd_service', 'device_group', @get_args)
+    end
+
+    def device_group=(val)
+      if val
+        @set_args[:state] = ''
+        @set_args[:dg] = val
+        config_set('itd_service', 'device_group', @set_args)
+      else
+        @set_args[:state] = 'no'
+        @set_args[:dg] = device_group
+        config_set('itd_service', 'device_group', @set_args) if
+        device_group
+      end
+      set_args_keys_default
+    end
+
+    def default_device_group
+      config_get_default('itd_service', 'device_group')
+    end
+
+    def exclude_access_list
+      config_get('itd_service', 'exclude_access_list', @get_args)
+    end
+
+    def exclude_access_list=(val)
+      if val
+        @set_args[:state] = ''
+        @set_args[:al] = val
+        config_set('itd_service', 'exclude_access_list', @set_args)
+      else
+        @set_args[:state] = 'no'
+        @set_args[:al] = exclude_access_list
+        config_set('itd_service', 'exclude_access_list', @set_args) if
+        exclude_access_list
+      end
+      set_args_keys_default
+    end
+
+    def default_exclude_access_list
+      config_get_default('itd_service', 'exclude_access_list')
+    end
+
+    def failaction
+      config_get('itd_service', 'failaction', @get_args)
+    end
+
+    def failaction=(state)
+      no_cmd = (state ? '' : 'no')
+      @set_args[:state] = no_cmd
+      config_set('itd_service', 'failaction', @set_args)
+      set_args_keys_default
+    end
+
+    def default_failaction
+      config_get_default('itd_service', 'failaction')
+    end
+
+    # this is an array like:
+    # [['ethernet 1/1', '1.1.1.1'], ['port-channel 100', '2.2.2.2'],
+    # ['vlan 2', '3.3.3.3']]
+    # show command output is like: Eth1/1, Po100, Vlan2
+    # so translate back to the input format
+    def ingress_interface
+      list = config_get('itd_service', 'ingress_interface', @get_args)
+      list.each do |intf, _next_hop|
+        intf[/Eth/] = 'ethernet ' if intf.include?('Eth')
+        intf[/Po/] = 'port-channel ' if intf.include?('Po')
+        intf[/Vlan/] = 'vlan ' if intf.include?('Vlan')
+      end
+      list
+    end
+
+    # only one next-hop is allowed per interface but
+    # due to nxos issues, it allows more than one;
+    # so the workaround is to clean up the current ingress
+    # intf and configure all of them from the manifest
```

Comment:
Don't mention manifest in the shared object since this will also by use by other orchestration agents like chef.